phases/bitrise_yml.go

```diff
@@ -55,16 +54,16 @@ func checkBranch(inputReader io.Reader) error {
 	return nil
 }
 
-// ParseDSLFile parses a bitrise.yml and returns a data model
-func ParseDSLFile(input io.Reader) (models.BitriseDataModel, error) {
-	var decodedDSL models.BitriseDataModel
-	if err := yaml.NewDecoder(input).Decode(&decodedDSL); err != nil {
+// ParseBitriseYMLFile parses a bitrise.yml and returns a data model
+func ParseBitriseYMLFile(input io.Reader) (models.BitriseDataModel, error) {
```

Comment:
NOTE: what do you think about using the parser from the bitrise cli: https://github.com/bitrise-io/bitrise/blob/master/cli/run_util.go#L198

It has the most recent validation for the bitrise yml, it also presents warnings if any.